mkosi/sysupdate.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import sys
+from pathlib import Path
+
+from mkosi.config import Args, Config, OutputFormat
+from mkosi.log import die
+from mkosi.run import run
+from mkosi.types import PathString
+
+
+def run_sysupdate(args: Args, config: Config) -> None:
+    if config.output_format != OutputFormat.disk:
+        die(f"{config.output_format} format is not supported by mkosi sysupdate")
+
+    if not config.split_artifacts:
+        die("SplitArtifacts= must be enabled to be able to use mkosi sysupdate")
+
+    if not config.sysupdate_dir:
+        die("No sysupdate definitions directory specified",
+            hint="Specify a directory containing systemd-sysupdate transfer definitions with SysupdateDirectory=")
+
+    if not (sysupdate := config.find_binary("systemd-sysupdate", "/usr/lib/systemd/systemd-sysupdate")):
+        die("Could not find systemd-sysupdate")
+
+    cmd: list[PathString] = [
+        sysupdate,
+        "--definitions", config.sysupdate_dir,
+        "--transfer-source", config.output_dir_or_cwd(),
+        *args.cmdline,
+    ]
+
+    run(
+        cmd,
+        stdin=sys.stdin,
+        stdout=sys.stdout,
+        env=os.environ | config.environment,
+        log=False,
+        sandbox=config.sandbox(
+            binary=sysupdate,
+            devices=True,
+            network=True,
```

Comment:
Do we need to allow network here? I'd gather that in the mkosi case everything will be local?

Reply:
Not necessarily, you have to do `PathRelativeTo=explicit` to look things up relative to the mkosi output directory so I'd prefer also allowing definitions that look up stuff remotely if that's what users want.